### ISSUE TITLE
Need a little help with a mod idea... (client-side commands)

### ISSUE THREAD

**Issue by BigPanda97** [Defective4/Another-Minecraft-Chat-Client#14]
I'm currently trying to make some kind of mod/plugin for this chat client.
However, I need to be able to make some kind of command handler for client-side commands.
(For example: “/mod” etc.)
As far as I know, this chat client does not have something like client-side commands yet, so I have to do it on my own.

But therefore, I need to be able to do two things:
1. Intercept messages which start with a client-side command
2. Be able to display the output of the command in the chat (client-side), without sending it to the server.

So basically, into which methods of the code would I have to “hook” to be able to implement such a feature?

**Comment by BigPanda97** [comment]
Thanks for the explanation! I will try to get it working using the “dirty” way first. But some “official” interfaces might be a good idea for the next version, especially if there are maybe other developers too in the future who perhaps might want to interact with the chat messages too.

**Comment by BigPanda97** [comment]
> But some “official” interfaces might be a good idea for the next version, especially if there are maybe other developers too in the future who perhaps might want to interact with the chat messages too.

Did you already add that? It would be nice to have something like an “official” API for this, so I don't have to “rebuild” everything in the “dirty” way in each version.

**Comment by Defective4** [comment]
@BigPanda97
Public API is still a Work In Progress and I plan to include it in the next release along with JavaDocs.
Also, thank you for you interest in my project!

By the way, out of curiosity, may I ask what kind of mod are you going to make?

**Comment by BigPanda97** [comment]
> @BigPanda97 Public API is still a Work In Progress and I plan to include it in the one of the upcoming releases along with JavaDocs. Also, thank you for your interest in my project!

Perfect, thanks!

> By the way, out of curiosity, may I ask what kind of mod are you going to make?

I'm playing around with several ideas I had in the past, but one of them would be something like [this](https://www.curseforge.com/minecraft/mc-mods/chatencryption) to make the private messages (/msg) more secure. ^^

**Comment by Defective4** [comment]
Hello @BigPanda97!
The first version of the public plugin API is now available!
I will soon start writing a wiki with guidelines, examples and tutorials, but until then you can use the generated javadocs.
If you are still interested in creating plugins for AMCC, you can download the Plugin API binaries and javadocs on the [releases](https://github.com/Defective4/Another-Minecraft-Chat-Client/releases).
Please let me know if you find this helpful!

**Comment by BigPanda97** [comment]
Hello @Defective4! Thanks, that's really really nice. I have some exams in the upcoming weeks, so it will take some time until I can test everything, but it is really cool that you started the developing of this awesome tool again!

**Comment by Defective4** [comment]
No problem!
Let me know if you need help since the wiki and documentation are not quite complete.
Also good luck on your exams @BigPanda97!
